File: frontend/src/types/competition.ts
export type CompetitionStatus = 'active' | 'ended' | 'cancelled'

export interface Competition {
  id: number
  venueId: number
  venueName: string
  slotId: number
  date: string
  time: string
  currentPrice: number
  bestPrice: number
  bidsCount: number
  timeLeft: string
  status: CompetitionStatus
  createdAt: string
  expiresAt: string
}

export interface CompetitionCreate {
  slotId: number
  venueId: number
  offeredPrice: number
}

export interface CompetitionBid {
  slotId: number
  offeredPrice: number
}

export interface CompetitionBidResponse {
  id: number
  competitionId: number
  venueId: number
  offeredPrice: number
  createdAt: string
}

File: backend/app/schemas/competition.py
from pydantic import BaseModel
from datetime import datetime
from enum import Enum

class CompetitionStatus(str, Enum):
    ACTIVE = "active"
    WON = "won"
    LOST = "lost"
    EXPIRED = "expired"

class CompetitionCreate(BaseModel):
    slot_id: int
    offered_price: int

class CompetitionBid(BaseModel):
    offered_price: int

class CompetitionResponse(BaseModel):
    id: int
    slot_id: int
    venue_id: int
    offered_price: int
    status: CompetitionStatus
    expires_at: datetime
    created_at: datetime
    
    class Config:
        from_attributes = True
